lib/wasm/WebAssemblyGenerator.js

```diff
@@ -181,13 +189,63 @@ const rewriteImportedGlobals = state => bin => {
 	return addWithAST(state.ast, bin, newGlobals);
 };
 
+/**
+ * Rewrite the export names
+ * @param {Object} state state
+ * @param {Object} state.ast Module's ast
+ * @param {Object} state.module Module
+ * @returns {ArrayBufferTransform} transform
+ */
+const rewriteExportNames = ({ ast, module }) => bin => {
+	return editWithAST(ast, bin, {
+		ModuleExport(path) {
+			const usedName = module.isUsed(path.node.name);
+			if (usedName) {
+				path.node.name = usedName;
+				// TODO remove this when fixed in @webassemblyjs
+				path.node.descr.id = t.numberLiteral(+path.node.descr.id.raw);
+			} else {
+				path.remove();
```

Comment:
This only removes a couple of bytes from the export section. We can implement proper func elimination in the future , I'ill work on that.

Reply:
This is all work we need to do at this place. DCE will be the job of the WASM minimizer (to be defined/implemented), which will operate on asset level.

Reply:
Ok, sounds good. I have a couple of wasm optimizations that we could put there as well.